.bazelrc

```diff
@@ -89,8 +89,8 @@ build:secure --config=relro
 build:release --compilation_mode opt
 build:release --config=secure
 build:release --copt -mllvm --copt -inline-threshold=2500
-build:release --linkopt=-flto
 build:release --@seastar//:stack_guards=False
+build:release --copt -flto=thin --copt -ffat-lto-objects
```

Comment:
So basically in release we will build every binary 2x: the non-PIC and PIC version of every .o, and both of these will be fat and then the non-PIC binary (redpanda) will be build using the bitcode side of the non-PIC .o, and then PIC binaries (tests) will be build using the binary side of the PIC .o files?

So half of everything would be never used: but we can't just build non-PIC-bitcode-only + PIC-binary-only, "because bazel"?

Reply:
Remember that Bazel builds lazily, so if the end target you're building doesn't need one version of the library it shouldn't build it. We happen to build both the binary and tests which means we double compile

Reply:
I mention PIC and non-PIC because which side is actually used in the link (binary or bitcode) is correlated with PIC + non-PIC so it seems almost possible to DTRT here but not sure if Bazel will let us.

Reply:
I see yea I don't think there is a way around this, but I can dig a little bit...

Reply:
From various chats in p-bazel I had the impression we want to eventually build all binaries the same in release mode (or at least the microbenchmarks)?

Note that the fat-lto compilation isn't really two compilations. In terms of compile time it's maybe 20-30% slower compared to compiling without `-fat-lto-objects` from a quick test.

Reply:
Yeah I guess this is OK, and right it's a doubling of "output" but not of time (but output size also matters due to gigantic nature of our objects). More verbalizing the complaint than expecting any solution here.

> From various chats in p-bazel I had the impression we want to eventually build all binaries the same in release mode (or at least the microbenchmarks)?

The same being no-PIC, right?

The main problem I think is the massive size of this requires, but I guess it's a good idea for release only? Plus we would get some of the disk space back by not needing the PIC side .o files at all.